src/app/common/modal/container-modal/container-modal.component.ts

```diff
@@ -0,0 +1,84 @@
+import {
+	AfterViewInit,
+	Component,
+	ComponentFactoryResolver,
+	Type,
+	ViewChild,
+	ViewContainerRef
+} from '@angular/core';
+
+import { untilDestroyed, UntilDestroy } from '@ngneat/until-destroy';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { Subject } from 'rxjs';
+import { AbstractModalDirective } from '../abstract-modal.directive';
+import { AbstractModalizedDirective } from '../abstract-modalized.directive';
+
+@UntilDestroy()
+@Component({
+	templateUrl: 'container-modal.component.html'
+})
+export class ContainerModalComponent extends AbstractModalDirective implements AfterViewInit {
+	@ViewChild('modalizedComponentContainer', { read: ViewContainerRef })
+	modalizedComponentContainer: ViewContainerRef;
+
+	modalizedComponent: Type<AbstractModalizedDirective>;
+
+	modalizedComponentProperties: { [key: string]: any } = {};
+
+	okSubject: Subject<void>;
+
+	cancelSubject: Subject<void>;
+
+	isOkDisabled = false;
+
+	constructor(
+		public modalRef: BsModalRef,
+		private componentFactoryResolver: ComponentFactoryResolver
+	) {
+		super(modalRef);
+	}
+
+	ngAfterViewInit(): void {
+		// Add the supplied modalized component to the content container
+		this.modalizedComponentContainer.clear();
+		const componentFactory = this.componentFactoryResolver.resolveComponentFactory(
+			this.modalizedComponent
+		);
+		const componentRef = this.modalizedComponentContainer.createComponent(componentFactory);
+
+		// Set any supplied component properties
+		Object.entries(this.modalizedComponentProperties).forEach(([key, value]) => {
+			Object.defineProperty(componentRef.instance, key, {
```

Comment:
I had initially just been doing these assignments with `componentRef.instance[key] = value`, but that was throwing a linting error about being unable to index Type AbstractModalizableDirective by string. I believe this accomplishes the same thing.